### PR TITLE
Compat update for String rename

### PR DESCRIPTION
**Pull request by StefanKarpinski**

### CODE REVIEW

File: test/test4d.jl [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import Images
-using Compat
```
Comment (yuyichao) [inline test/test4d.jl]: Unnecessary

File: src/contrast.jl [hunk 2]
```diff
@@ -6,7 +6,7 @@ using Tk
 using Winston
 using Images
 import Colors.Fractional
-using Compat
```
Comment (yuyichao) [inline src/contrast.jl]: Unnecessary

File: src/annotations.jl [hunk 3]
```diff
@@ -48,13 +48,13 @@ type AnnotationText
...
+    fontfamily::String
+    fontoptions::String
     fontsize::Integer
-    fontdesc::ASCIIString
+    fontdesc::String
     angle::Float64
-    halign::ASCIIString
-    valign::ASCIIString
+    halign::String
+    valign::String
```
Comment (yuyichao) [inline src/annotations.jl]: Are these fields never returned through any of the public API's?

Comment (timholy) [inline src/annotations.jl]: Basically, no. There's just passed to Cairo.jl.

**Comment by timholy** [comment]
Thanks!

**Comment by yuyichao** [comment]
This is still adding unnecessary code = = .....

**Comment by timholy** [comment]
Sure, but one can think of it as a preventative measure.

**Comment by yuyichao** [comment]
It's probably fine to keep. I always find it confusing to see unnecessary code without explanation though. Figuring out why such code is added and if it can be deleted at a later time often involving tracing through the git history and guessing what the author meant. In this case the comment on the PR should at least make the guessing part unnecessary.

**Comment by StefanKarpinski** [comment]
@yuyichao, I would argue that `using Compat` should now be spelled `using Compat; import Compat.String` universally.

**Pull request merged**